xpublish_edr/formats/to_geojson.py

```diff
@@ -0,0 +1,33 @@
+"""
+Generate GeoJSON responses for an xarray dataset for EDR queries
+"""
+
+import geopandas as gpd
+import xarray as xr
+from fastapi import Response
+
+
+def to_geojson(ds: xr.Dataset):
+    """Return a GeoJSON response from an xarray dataset"""
+    ds = ds.squeeze()
+    x_col = ds.cf["X"].name
+    y_col = ds.cf["Y"].name
+    if "T" in ds.cf:
+        time_col = ds.cf["T"].name
+    else:
+        time_col = None
```

Comment:
This will do the same in only one pass over the Dataset and no overhead in constructing dataarrays.
```suggestion
    coordinates = ds.cf.coordinates
    x_col,  = coordinates["X"]
    y_col, = coordinates["Y"]
    time_col = coordinates.get("T")
```

Later we should profile this stuff and see if it matters once network latency is factored in.